src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2594,27 +2594,21 @@ private function registerNotifierConfiguration(array $config, ContainerBuilder $
 
         if (ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, $parentPackages, true) && ContainerBuilder::willBeAvailable('symfony/mercure-bundle', MercureBundle::class, $parentPackages, true) && \in_array(MercureBundle::class, $container->getParameter('kernel.bundles'), true)) {
             $container->getDefinition($classToServices[MercureTransportFactory::class])
-                ->replaceArgument('$registry', new Reference(HubRegistry::class))
-                ->replaceArgument('$client', new Reference('http_client', ContainerBuilder::NULL_ON_INVALID_REFERENCE))
-                ->replaceArgument('$dispatcher', new Reference('event_dispatcher', ContainerBuilder::NULL_ON_INVALID_REFERENCE));
+                ->replaceArgument('$registry', new Reference(HubRegistry::class));
```

Comment:
can't we use index_foo instead of named args? that's what I meant :)

Reply:
I didn't touch this line as it already has been like this before #52998

Reply:
I would still remove it, core shouldn't use reflection to do it's wiring. We don't do this anywhere else but in one case where it's on purpose. Here, it's just by laziness :)